lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -227,74 +204,82 @@ Point TruncatedDistribution::getRealization() const
   for (;;)
   {
     Point realization(distribution_.getRealization());
-    if ((!finiteLowerBound_ || realization[0] >= lowerBound_) && (!finiteUpperBound_ || realization[0] <= upperBound_)) return realization;
+    if (bounds_.contains(realization)) return realization;
   }
 }
 
 /* Get the DDF of the distribution: DDF_trunc = 1[a, b] * DDF / (CDF(b) - CDF(a)) */
 Point TruncatedDistribution::computeDDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
 
-  const Scalar x = point[0];
-  if ((x <= lowerBound_) || (x > upperBound_)) return Point(1, 0.0);
+  if (!bounds_.contains(point)) return Point(getDimension(), 0.0);
   return normalizationFactor_ * distribution_.computeDDF(point);
 }
 
 
 /* Get the PDF of the distribution: PDF_trunc = 1[a, b] * PDF / (CDF(b) - CDF(a)) */
 Scalar TruncatedDistribution::computePDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  Interval offset(Point(getDimension(), -quantileEpsilon_), Point(getDimension(), quantileEpsilon_));
+  if (!(bounds_ + offset).contains(point)) return 0.0;
 
-  const Scalar x = point[0];
-  if ((x < lowerBound_ - quantileEpsilon_) || (x > upperBound_ + quantileEpsilon_)) return 0.0;
   return normalizationFactor_ * distribution_.computePDF(point);
 }
 
 
 /* Get the CDF of the distribution: CDF_trunc = 1[a, b] * (CDF - CDF(a)) / (CDF(b) - CDF(a)) + 1]b, inf] */
 Scalar TruncatedDistribution::computeCDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+  {
+    const Scalar xK = point[k];
+    if (xK <= bounds_.getLowerBound()[k]) return 0.0;
+    if (xK >= bounds_.getUpperBound()[k]) return 1.0;
+  }
 
-  const Scalar x = point[0];
-  if (x <= lowerBound_) return 0.0;
-  if (x >= upperBound_) return 1.0;
   // If tail=true, don't call distribution_.computeCDF with tail=true in the next line!
   return normalizationFactor_ * (distribution_.computeCDF(point) - cdfLowerBound_);
```

Comment:
No. The CDF of a truncated distribution is equal to the probability of the interval [lower, point] divided by the probability of the truncation interval, ie distribution_.computeProbability(Interval(range_.getLowerBound(), point)) * normalizationFactor_ or marginal_.computeProbability(Interval(marginalLowerBound, marginalPoint)) * normalizationFactor_ if a part of the components of point are greater than their associated upper bounds.